packages/babel-helpers/src/helpers/applyDecs2305.ts

```diff
@@ -674,32 +189,407 @@ function defineMetadata(Class: any, metadata: any) {
+    for (var i = 0; i < initializers.length; i++) {
+      initializers[i].call(value);
+    }
+    return value;
+  }
+
+  function assertCallable(
+    fn: any,
+    hint1: string,
+    hint2?: string,
+    throwUndefined?: boolean,
+  ) {
+    if (typeof fn !== "function") {
+      if (throwUndefined || fn !== void 0) {
+        throw new TypeError(
+          hint1 +
+            " must " +
+            (hint2 || "be") +
+            " a function" +
+            (throwUndefined ? "" : " or void 0"),
+        );
+      }
+    }
+    return fn;
+  }
+
+  /* @no-mangle */
+  function applyDec(
+    Class: any,
+    decInfo: DecoratorInfo,
+    decoratorsHaveThis: number,
+    name: string,
+    kind: PROP_KIND,
+    metadata: any,
+    initializers: Function[],
+    ret?: Function[],
+    isStatic?: boolean,
+    isPrivate?: boolean,
+    isField?: boolean,
+    isAccessor?: boolean,
+    hasPrivateBrand?: Function,
+  ) {
+    function assertInstanceIfPrivate(target: any) {
+      if (!hasPrivateBrand(target)) {
+        throw new TypeError(
+          "Attempted to access private element on non-instance",
+        );
+      }
+    }
+
+    var decs = decInfo[0],
+      decVal = decInfo[3],
+      _: any,
+      isClass = !ret;
+
+    if (!isClass) {
+      if (!decoratorsHaveThis && !Array.isArray(decs)) {
+        decs = [decs];
+      }
+
+      var desc: PropertyDescriptor = {},
+        init: Function[] = [],
+        key: "get" | "set" | "value" =
+          kind === PROP_KIND.GETTER
+            ? "get"
+            : kind === PROP_KIND.SETTER || isAccessor
+              ? "set"
+              : "value";
+
+      if (isPrivate) {
+        if (isField || isAccessor) {
+          desc = {
+            get: setFunctionName(
+              function (this: any) {
+                return decVal(this);
+              },
+              name,
+              "get",
+            ),
+            set: function (this: any, value: any) {
+              decInfo[4](this, value);
+            },
+          };
+        } else {
+          desc[key] = decVal;
+        }
+
+        if (!isField) {
+          setFunctionName(
+            desc[key],
+            name,
+            kind === PROP_KIND.METHOD ? "" : key,
+          );
+        }
+      } else if (!isField) {
+        desc = Object.getOwnPropertyDescriptor(Class, name);
+      }
+    }
+
+    var newValue;
+
+    for (var i = decs.length - 1; i >= 0; i -= decoratorsHaveThis ? 2 : 1) {
+      var dec = (decs as Function[])[i],
+        decThis = decoratorsHaveThis ? (decs as any[])[i - 1] : void 0;
+
+      var decoratorFinishedRef: DecoratorFinishedRef = {};
+      var ctx: DecoratorContext = {
+        kind: [
+          "field",
+          "accessor",
+          "method",
+          "getter",
+          "setter",
+          "field",
+          "class",
+        ][kind] as any,
+
+        name: name,
+        metadata: metadata,
+      };
+
+      ctx.addInitializer = function (
```

Comment:
Why is this not inside the object literal above?

Reply:
Oops, I forgot this when rebasing https://github.com/babel/babel/pull/16132.

Reply:
Oops, I forgot this when rebasing https://github.com/babel/babel/pull/16132.